File: 01/solution1.c
#include <stdio.h>

int main(void)
{
    FILE* f = fopen("input.txt", "rb");
    int acc = 0;
    
    if (f == NULL) {
        fprintf(stdout, "Error: File could not be opened.");

        return 1;
    }

    do
    {
        int element = -1;
        if (fscanf(f, "%d", &element) == EOF)
            break;

        acc += element;
    } while(1);

    fprintf(stdout, "%d\n", acc);

    return 0;
}

File: 01/solution1.pas
program solution1;

uses
    SysUtils;

const
    FILE_NAME = 'input.txt';

var
    f: textfile;
    line: string;
    acc, element: Int64;
begin
    AssignFile(f, FILE_NAME);
    Reset(f);

    acc := 0;

    while not eof(f) do
    begin
        ReadLn(f, line);

        element := StrToInt(
            Copy(
                line,
                2,
                Length(line) - 1
            )
        );

        if line[1] = '-' then
            element := -element;

        acc := acc + element;
    end;

    WriteLn(acc);
end.
